docs/execution_providers/MKL-DNN-ExecutionProvider.md

```diff
@@ -1,10 +1,10 @@
-# MKL-DNN Execution Provider
+# DNNL Execution Provider
```

Comment:
Other files with mkl-dnn you renamed but not this and subgraph documentation?

Reply:
Yes. Wanted to check with you before I renaming them. There are some references to this web 
 page (ex. release notes) and wondering if it is ok to break it.

Reply:
> Yes. Wanted to check with you before I renaming them. There are some references to this web
> page (ex. release notes) and wondering if it is ok to break it.

Better not rename it then. It's fine. The mkldnn GitHub url itself was not renamed right.
Please fix the typo then in BUILD.md then